lib/ringsc.gi

```diff
@@ -917,11 +917,25 @@ end);
 InstallOtherMethod(IsUnit,"for finite Rings",
   IsCollsElms,[IsRing,IsScalar],0,
 function(R,e)
-  if not IsFinite(R) then
+local o,pow,a;
+  if not IsFinite(R) or One(R)=fail then
     TryNextMethod();
   fi;
-  return One(R)<>fail
-         and ForAny(Enumerator(R),x->x*e=One(R) and e*x=One(R));
+
+  # compute powers of e until we reach one or repeat
```

Comment:
The previous code checked whether there is an element that is both a left and a right inverse.
The proposed code checks for a power that is the identity element, which is sufficient if the multiplication is associative, but why is it correct in nonassociative situations? The code establishes a left inverse, but why is it a right inverse?

Reply:
Ah. My world is associative. Will fix.

Reply:
Fixed in original